app/src/main/java/com/ivy/wallet/RootActivity.kt

```diff
@@ -465,4 +469,78 @@ class RootActivity : AppCompatActivity(), RootScreen {
         val addTransactionWidget = ComponentName(this, widget)
         appWidgetManager.requestPinAppWidget(addTransactionWidget, null, null)
     }
+
+}
+//For pushing Notification to the user as a reminder at every 12AM
+// urging to fill in expenses, transactions and incomes
+fun createNotificationChannel(context: Context) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val name = "DailyReminderChannel"
+        val descriptionText = "Channel for daily reminders"
+        val importance = NotificationManager.IMPORTANCE_HIGH
+        val channel = NotificationChannel("DAILY_REMINDER_CHANNEL_ID", name, importance).apply {
+            description = descriptionText
+        }
+        val notificationManager: NotificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.createNotificationChannel(channel)
+    }
 }
+fun sendNotification(context: Context) {
+    val intent = Intent(context, MainActivity::class.java).apply {
+        flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+    }
+    val pendingIntent: PendingIntent = PendingIntent.getActivity(context, 0, intent,
+        PendingIntent.FLAG_IMMUTABLE)
+
+    val builder = NotificationCompat.Builder(context, "DAILY_REMINDER_CHANNEL_ID")
+        .setSmallIcon(R.drawable.ic_launcher-playstore)
+        .setContentTitle("EveryDay Reminder")
+        .setContentText("Fill in todays expences,transactions and incomes.")
+        .setPriority(NotificationCompat.PRIORITY_HIGH)
+        .setContentIntent(pendingIntent)
+        .setAutoCancel(true)
+
+    with(NotificationManagerCompat.from(context)) {
+        if (ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.POST_NOTIFICATIONS
+            ) != PackageManager.PERMISSION_GRANTED
```

Comment:
The code would simpler if:
1. Extract the permission check as a `hasNotificationsPermission` function
2. if (hasNotificationsPermisson()) notify